### PR TITLE
Fix qtile-cmd failure, #1410.

### PR DESCRIPTION
**Pull request by whatacold**
This fixes the failure of qtile-cmd as described in #1410, it excludes the initial object of `CommandGraphRoot` type, which has no meaningful children in qtile manager.

### CODE REVIEW

**Comment by whatacold** [comment]
It seems that changing the type hint of the `name` argument of `InteractiveCommandClient.__getitem__()` from `str` to `Union[str, int]` involves many other signature changes such as those in `def navigate()`, `def has_item()`, etc.

Should I proceed or leave it as is in this PR?

**Comment by whatacold** [comment]
> Should I proceed or just leave it as is in this PR?

Well, I think I'd better keep it as is.

**Comment by ramnes** [comment]
Also related to #1298 right?

File: libqtile/command_client.py [hunk 1]
```diff
@@ -220,23 +220,42 @@ def __getitem__(self, name: str) -> "InteractiveCommandClient":
...
+                raise SelectError("Item not available in object",
+                                  str(item), self._current_node.selectors)
+
+        if isinstance(name, str) and name.isdigit():
+            # Check the item as is, and check its int version once more if it fails
+            try:
+                _check_item(name)
+            except SelectError:
+                name = int(name)
+                _check_item(name)
```
Comment (ramnes) [inline libqtile/command_client.py]: I really don't like this. We should have some way to know whether we must send a string or int.

File: test/test_qtile_cmd.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+
+server_config = pytest.mark.parametrize("qtile", [ServerConfig], indirect=True)
+
+
+@server_config
+def test_qtile_cmd(qtile):
+
+    def _qtile_cmd(obj_spec, function, args):
+        "Imitate the behavior of qtile_cmd"
```
Comment (ramnes) [inline test/test_qtile_cmd.py]: Is it impossible to use `qtile_cmd` directly?

File: libqtile/command_client.py (hunk 1 above)
Comment (whatacold) [reply libqtile/command_client.py]: > Also related to #1298 right?

Yeah!

How about normalizing the item at first? There is no way to determine if an item is int or str from shell arguments.

File: test/test_qtile_cmd.py (hunk 2 above)
Comment (whatacold) [reply test/test_qtile_cmd.py]: You mean by calling the script directly? It's possible but I'm afraid it's a bit slower, and the output need to be parsed again if some fields need to be asserted.

File: libqtile/command_client.py (hunk 1 above)
Comment (ramnes) [reply libqtile/command_client.py]: From shell arguments no, but from functions parameters yes!

File: test/test_qtile_cmd.py (hunk 2 above)
Comment (ramnes) [reply test/test_qtile_cmd.py]: Yes but then we test the whole thing and I feel it has a lot of value, more than the speed gain.

Comment (whatacold) [reply test/test_qtile_cmd.py]: Yeah, I agree that the correctness of the WM and also the test is more important, so I changed it to call `qtile-cmd` directly now.

File: libqtile/command_client.py (hunk 1 above)
Comment (whatacold) [reply libqtile/command_client.py]: Now it's normalized in `qtile_cmd.py`, only where it's necessary.

File: test/test_command.py [hunk 3]
```diff
@@ -161,15 +161,6 @@ class ServerConfig:
 server_config = pytest.mark.parametrize("qtile", [ServerConfig], indirect=True)
 
 
-@server_config
-def test_item_types(qtile):
```
Comment (tych0) [inline test/test_command.py]: Can we change this to use the new normalize_item() instead of just dropping the test?

**Comment by flacjacket** [comment]
It looks like all of the comments on this have been resolved, and I think this looks great now, thanks!

**Pull request merged**